src/main/java/seedu/address/logic/commands/volunteercommands/FindVolunteerCommand.java

```diff
@@ -47,18 +45,14 @@ public CommandResult execute(Model model) throws CommandException {
             throw new CommandException(String.format(MESSAGE_VOLUNTEER_NOT_FOUND, searchString));
         }
 
-        Predicate<Volunteer> volunteerContainsSearchString = volunteer ->
-                volunteer.getName().toString().toLowerCase().contains(searchString.toLowerCase());
-        model.updateFilteredVolunteerList(volunteerContainsSearchString);
+        ObservableList<Volunteer> filteredEvents = model.filterVolunteersByName(searchString);
 
-        ObservableList<Volunteer> filteredVolunteers = model.getFilteredVolunteerList();
-        if (filteredVolunteers.isEmpty()) {
+        if (filteredEvents.isEmpty()) {
             model.updateFilteredVolunteerList(PREDICATE_SHOW_ALL_VOLUNTEERS);
             return new CommandResult(String.format(MESSAGE_VOLUNTEER_NOT_FOUND, searchString));
         }
```

Comment:
Can we do this inside model or the predicate instead? Imo its not ideal to retrieve the list of events and manipulate them in a command.